tests/src/analysis/CMakeLists.txt

```diff
@@ -61,27 +36,38 @@ ENDIF (APPLE)
 #No relinking and full RPATH for the install tree
 #See: http://www.cmake.org/Wiki/CMake_RPATH_handling#No_relinking_and_full_RPATH_for_the_install_tree
 
-MACRO (ADD_QGIS_TEST testname testsrc)
-  SET(qgis_${testname}_SRCS ${testsrc} ${util_SRCS})
-  SET(qgis_${testname}_MOC_CPPS ${testsrc})
-  ADD_EXECUTABLE(qgis_${testname} ${qgis_${testname}_SRCS})
-  SET_TARGET_PROPERTIES(qgis_${testname} PROPERTIES AUTOMOC TRUE)
-  TARGET_LINK_LIBRARIES(qgis_${testname}
+MACRO (ADD_QGIS_TEST TESTSRC)
+  SET (TESTNAME  ${TESTSRC})
+  STRING(REPLACE "test" "" TESTNAME ${TESTNAME})
+  STRING(REPLACE "qgs" "" TESTNAME ${TESTNAME})
+  STRING(REPLACE ".cpp" "" TESTNAME ${TESTNAME})
+  SET (TESTNAME  "qgis_${TESTNAME}test")
+
+  SET(${TESTNAME}_SRCS ${TESTSRC} ${util_SRCS})
+  SET(${TESTNAME}_MOC_CPPS ${TESTSRC})
+  ADD_EXECUTABLE(${TESTNAME} ${${TESTNAME}_SRCS})
+  SET_TARGET_PROPERTIES(${TESTNAME} PROPERTIES AUTOMOC TRUE)
+  TARGET_LINK_LIBRARIES(${TESTNAME}
     ${QT_QTCORE_LIBRARY}
     ${QT_QTTEST_LIBRARY}
     qgis_analysis)
-  ADD_TEST(qgis_${testname} ${CMAKE_CURRENT_BINARY_DIR}/../../../output/bin/qgis_${testname} -maxwarnings 10000)
+  ADD_TEST(${TESTNAME} ${CMAKE_CURRENT_BINARY_DIR}/../../../output/bin/${TESTNAME} -maxwarnings 10000)
```

Comment:
Could that be

    ADD_TEST(${TESTNAME} ${CMAKE_BINARY_DIR}/output/bin/${TESTNAME} -maxwarnings 10000)

I think sometimes abosolute is better than relative ;)